tests/unit/test_distro.py

```diff
@@ -346,6 +346,78 @@ def test_configure_env_endpoint(self, mocker):
         assert os.environ[OTEL_LOGS_EXPORTER] == "otlp"
         assert os.environ.get("OTEL_SEMCONV_STABILITY_OPT_IN") == "http"
 
+    def test_configure_env_service_key_only(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-01.cloud.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_only(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "apm.collector.na-02.cloud.solarwinds.com",
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-02.cloud.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_only_dev(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "apm.collector.na-01.dev-ssp.solarwinds.com",
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-01.dev-ssp.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_only_stg(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "apm.collector.na-02.st-ssp.solarwinds.com",
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-02.st-ssp.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_only_invalid(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "www.google.com",
```

Comment:
This one I am not sure about, and apologies because it's not in ticket OP either.

@cheempz , does this make sense for this iteration?: If `SW_APM_COLLECTOR` is anything but `apm.collector`, then APM uses the `otel.collector.na-01` default. OR, APM uses what's configured and lets settings collection and export fail for visible error.

Reply:
My thinking was to greedily extract `xx-yy.<cloud/st-ssp/dev-ssp>` from the environment variable and substitute to `https://otel.collector.<substitute>.solarwinds.com:443`

Let me know if it doesn't making sense. Thanks!

Reply:
yes it makes sense to use the fallback value if SW APM config for collector is anything other than `apm.collector.*`